src/components/_internal/BaseButton/BaseButton.tsx

```diff
@@ -52,6 +52,7 @@ const BaseButton: React.FC<
   isExpanded = false,
   loadingText = 'Loading',
   className,
+  ariaLabel,
```

Comment:
I think it would be better not to introduce a custom prop name and rather destructure `aria-label` prop from the `...props` and use it where we need. That way people wouldn't need to remember that button has a special name for this.